src/stream/src/executor_v2/lookup/impl_.rs

```diff
@@ -195,21 +205,35 @@ impl<S: StateStore> LookupExecutor<S> {
     /// messages until there's one.
     ///
     /// If we can use `async_stream` to write this part, things could be easier.
-    pub async fn next_inner(&mut self) -> Result<Option<Message>> {
-        match self.input.next().await.expect("unexpected end of stream")? {
-            ArrangeMessage::Barrier(barrier) => {
-                self.process_barrier(barrier.clone()).await?;
-                Ok(Some(Message::Barrier(barrier)))
+    #[try_stream(ok = Message, error = TracedStreamExecutorError)]
+    pub async fn execute_inner(mut self: Box<Self>) {
+        let input = std::mem::replace(&mut self.input, None);
```

Comment:
Use `std::mem::take` to avoid explicitly write a `None`. `take` will replace with the default value.